### PR TITLE
Added CloudPut methods/tests for client/rpc/storage/server

### PR DESCRIPTION
**Pull request by mikeynap**
Working on a project with Cisco and we needed the ability to add CloudConfig files over gRPC. 

Is this something that you would be comfortable allowing?

### CODE REVIEW

**Comment by mikeynap** [comment]
Which version of protoc is this using?

File: bootcfg/storage/testfakes/fixtures.go [hunk 1]
```diff
@@ -39,12 +39,23 @@ var (
 
 	// IgnitionYAMLName is an Ignition template name for testing.
 	IgnitionYAMLName = "ignition.tmpl"
-
+
```
Comment (danehans) [inline bootcfg/storage/testfakes/fixtures.go]: ^ looks like whitespace that needs to be removed.

File: bootcfg/storage/testfakes/fixtures.go [hunk 2]
```diff
@@ -39,12 +39,23 @@ var (
...
+	
 	// IgnitionYAML is an Ignition template for testing.
 	IgnitionYAML = `ignition_version: 1
 systemd:
   units:
     - name: etcd2.service
       enable: true
 `
+	// CloudYAMLName is an CloudConfig template name for testing.
+	CloudYAMLName = "ignition.tmpl"
```
Comment (danehans) [inline bootcfg/storage/testfakes/fixtures.go]: should this be cloud.tmpl to differentiate from the ignition template?

**Comment by dghubble** [comment]
Although Ignition is recommended, the `/cloud` endpoint is still supported and gRPC API coverage is in-scope. I'll take a look through after `v0.4.0` is squared away this week.

**Comment by dghubble** [comment]
I believe protoc-3.0.0-beta-2. This needs to be captured in the build process.

**Comment by dghubble** [comment]
I've tested `make tools` and `make codegen` against your PR. Once https://github.com/coreos/coreos-baremetal/pull/305 merges you should be able to run those re-generate code and tests should pass.  Thanks for highlighting the lack of a pinned protocol buffer build target.

**Pull request closed without merging**